sormas-api/src/main/java/de/symeda/sormas/api/sample/SampleDto.java

```diff
@@ -481,19 +481,26 @@ public static SampleDto buildReferral(UserReferenceDto userRef, SampleDto referr
 		} else {
 			sample = build(userRef, associatedEventParticipant);
 		}
-		sample.setSampleDateTime(referredSample.getSampleDateTime());
-		sample.setSampleMaterial(referredSample.getSampleMaterial());
-		sample.setSampleMaterialText(referredSample.getSampleMaterialText());
-		sample.setSampleSource(referredSample.getSampleSource());
-		sample.setPathogenTestingRequested(referredSample.getPathogenTestingRequested());
-		sample.setAdditionalTestingRequested(referredSample.getAdditionalTestingRequested());
-		sample.setRequestedPathogenTests(referredSample.getRequestedPathogenTests());
-		sample.setRequestedAdditionalTests(referredSample.getRequestedAdditionalTests());
-		sample.setPathogenTestResult(PathogenTestResultType.PENDING);
+		migrateAttributesOfPhysicalSample(referredSample, sample);
 
 		return sample;
 	}
 
+	public static void migrateAttributesOfPhysicalSample(SampleDto source, SampleDto target) {
```

Comment:
I don't really see a reason to have this function extracted from the above one. At least is must not be public and/or maybe a better name, I don't really see the meaning of physicalSample on the context of source and target, which one is the physical one and why is it physical. Maybe `migrateAttributesOfReferredSample(source, referred)` would be more meaningful.

Reply:
The `public` declaration is a leftover from a state where attributes should sometimes be migrated from the lab mesage instead of the sample. I will change that.

The physical sample is neither the source, nor the target. This method is about migrating the attributes that belong to the real (physical) sample out there in the labs. Source and target both have attributes that somehow have to do with the physical sample, but may vary. For example, the `specimenCondition` may be different in source and target. In one lab (source), the `specimenCondition` may be `ADEQUATE`. But then during transport to another lab (target) the `specimenCondition` might change to `NOT_ADEQUATE`.

In contrast, the attributes of the physical sample don't change (e.g. `samplingReason`) and thus should be migrated when a sample referral is created in SORMAS.

I extracted this function to make it more visible. In the past, adding attributes in the migration was often forgotten (that's why I included `fieldSampleId`, `samplingReason` etc. into this method.

So are you okay with making the method private and adding a comment more or less explaining the points above?

Reply:
okay